queries.sql

```diff
@@ -58,4 +58,127 @@ FROM
     animals
 WHERE
     (weigth_kg >= 10.4)
-    AND (weigth_kg <= 17.3);
\ No newline at end of file
+    AND (weigth_kg <= 17.3);
+
+begin;
+
+UPDATE
+    animals
+SET
+    species = 'unspecified';
+
+select
+    *
+from
+    animals;
+
+rollback;
+
+select
+    *
+from
+    animals;
+
+begin;
+
+UPDATE
+    animals
+SET
+    species = 'digimon'
+WHERE
+    name LIKE '%mon';
+
+select
+    *
+from
+    animals;
+
+UPDATE
+    animals
+SET
+    species = 'pokemon'
+WHERE
+    species IS NULL
+    OR species = '';
+
+select
+    *
+from
+    animals;
+
+commit;
+
+select
+    *
+from
+    animals;
+
+SELECT
+    COUNT(*)
+FROM
+    animals;
+
+SELECT
+    COUNT(*)
+FROM
+    animals
+WHERE
+    (escape_attempts = 0);
+
+SELECT
+    AVG(weigth_kg)
+from
+    animals;
+
+SELECT
+    SUM(escape_attempts)
+FROM
+    animals
+WHERE
+    (neutered = true);
+
+SELECT
+    SUM(escape_attempts)
+FROM
+    animals
+WHERE
+    (neutered = false);
```

Comment:
- [ ] Really smart answering the question **who escapes the most, neutered or not neutered** in two steps :clap: However this could be done in only one step by making use of `GROUP BY`. Therefore I will suggest you refactor your code to include the `GROUP BY` key.
HIT: Your code could look like this :point_down: 
```sql
Who escapes the most, neutered or not neutered animals?

   SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```

or
```sql
  SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;
```